src/System.Data.SqlClient/src/System/Data/SqlTypes/SQLMoney.cs

```diff
@@ -171,16 +171,12 @@ public long ToInt64()
                 throw new SqlNullValueException();
 
             long ret = _value / (s_lTickBase / 10);
-            bool fPositive = (ret >= 0);
             long remainder = ret % 10;
             ret = ret / 10;
 
             if (remainder >= 5)
             {
-                if (fPositive)
-                    ret++;
-                else
-                    ret--;
+                ret++;
```

Comment:
If (ret % 10) >= 5, then by definition fPositive would be true as ret would have to be positive.